File: ServiGo/backend/apps/usuarios/routes/reserva_routes.py
from django.urls import path

from ..controllers.reserva_controller import (
    CrearReservaView,
    ListaReservasView,
    DetalleReservaView,
    ConfirmarReservaView,
    CancelarReservaView,
    ReprogramarReservaView,
    HistorialReservasView,
    DisponibilidadView,
    RechazarReservaView,
    HistorialClienteView,
)

urlpatterns = [
    path("reservas/", CrearReservaView.as_view(), name="crear-reserva"),
    path("reservas/lista/", ListaReservasView.as_view(), name="lista-reservas"),
    path("reservas/<int:pk>/", DetalleReservaView.as_view(), name="detalle-reserva"),
    path("reservas/<int:pk>/confirmar/", ConfirmarReservaView.as_view(), name="confirmar-reserva"),
    path("reservas/<int:pk>/cancelar/", CancelarReservaView.as_view(), name="cancelar-reserva"),
    path("reservas/<int:pk>/reprogramar/", ReprogramarReservaView.as_view(), name="reprogramar-reserva"),
    path("reservas/historial/", HistorialReservasView.as_view(), name="historial-reservas"),
    path("reservas/disponibilidad/", DisponibilidadView.as_view(), name="disponibilidad"),
    path("reservas/<int:pk>/rechazar/", RechazarReservaView.as_view(), name="rechazar-reserva"),
    path("profesional/<int:profesional_id>/clientes/<int:cliente_id>/historial/", HistorialClienteView.as_view(), name="historial-cliente"),
]

File: backend/src/routes/reserva.routes.ts
import { Router } from "express";
import { ReservaController } from "../controllers/reserva.controller";
import { authenticateToken } from "../middlewares/auth.middleware";
import { authorizeRoles } from "../middlewares/role.middleware";
import { ROLES } from "../data/roles";

const router = Router();

router.use(authenticateToken);

router.post(
  "/reservas/",
  authorizeRoles(ROLES.CLI, ROLES.ADMIN),
  ReservaController.crear
);

router.get(
  "/reservas/lista/",
  ReservaController.listar
);

router.get(
  "/reservas/historial/",
  ReservaController.historial
);

router.get(
  "/reservas/disponibilidad/",
  ReservaController.disponibilidad
);

router.get(
  "/reservas/:id/",
  ReservaController.detalle
);

router.patch(
  "/reservas/:id/confirmar/",
  authorizeRoles(ROLES.PRO, ROLES.ADMIN),
  ReservaController.confirmar
);

router.patch(
  "/reservas/:id/completar/",
  authorizeRoles(ROLES.PRO, ROLES.ADMIN),
  ReservaController.completar
);

router.patch(
  "/reservas/:id/cancelar/",
  authorizeRoles(ROLES.CLI, ROLES.PRO, ROLES.ADMIN),
  ReservaController.cancelar
);

router.patch(
  "/reservas/:id/reprogramar/",
  authorizeRoles(ROLES.CLI, ROLES.PRO, ROLES.ADMIN),
  ReservaController.reprogramar
);

router.patch(
  "/reservas/:id/rechazar/",
  authorizeRoles(ROLES.PRO, ROLES.ADMIN),
  ReservaController.rechazar
);

router.get(
  "/profesional/:profesionalId/clientes/:clienteId/historial/",
  authorizeRoles(ROLES.PRO, ROLES.ADMIN),
  ReservaController.historialCliente
);

export default router;